### PR TITLE
Support generic_xds_config in GCE/GKE frameworks and Python tests

### PR DESCRIPTION
**Pull request by lidizheng**
As title. This PR blocks https://github.com/grpc/grpc/pull/27794. Related https://github.com/grpc/proposal/pull/267.

The difference between new CSDS and old CSDS, see [Config Dump Design: GenericXdsConfig vs. PerXdsConfig
](https://github.com/grpc/proposal/blob/master/A40-csds-support.md#config-dump-design-genericxdsconfig-vs-perxdsconfig).

### CODE REVIEW

File: src/python/grpcio_tests/tests/csds/test_csds.py [hunk 1]
```diff
@@ -85,7 +85,10 @@ def get_xds_config_dump(self):
     def test_has_node(self):
         resp = self.get_xds_config_dump()
         self.assertEqual(1, len(resp.config))
-        self.assertEqual(4, len(resp.config[0].xds_config))
+        self.assertEqual(
```
Comment (gnossen) [inline src/python/grpcio_tests/tests/csds/test_csds.py:88]: I'm trying to understand this change. I don't see an implementation change for csds. Previously, we asserted that `len(resp.config[0].xds_config)` was 4. Doesn't that imply that `resp.config[0].generic_xds_config` is empty?

Are entries in `xds_config` mutually exclusive with those in `generic_xds_config`? That is, if an LDS entry shows up in `generic_xds_config`, is it guaranteed that it won't also show up in `xds_config`?

File: tools/run_tests/run_xds_tests.py [hunk 2]
```diff
@@ -2304,6 +2305,39 @@ def test_csds(gcp, original_backend_service, instance_group, server_uri):
...
+                        listener = generic_xds_config["xds_config"]
+                        if listener['name'] != server_uri:
+                            logger.info('Invalid Listener name %s != %s',
+                                        listener_name, server_uri)
+                            ok = False
+                    elif re.search(r'\.RouteConfiguration$',
+                                   generic_xds_config['typeUrl']):
+                        seen.add('rds')
+                        route_config = generic_xds_config["xds_config"]
+                        if len(route_config['virtual_hosts']) <= 0:
```
Comment (gnossen) [inline tools/run_tests/run_xds_tests.py:2320]: Nit: How could the length be less than 0?

File: tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_testcase.py [hunk 3]
```diff
@@ -107,6 +107,21 @@ def __init__(self, xds_json: JsonType):
...
+                elif re.search(r'\.RouteConfiguration$',
+                               generic_xds_config['typeUrl']):
+                    self.rds = generic_xds_config["xdsConfig"]
+                elif re.search(r'\.Cluster$', generic_xds_config['typeUrl']):
+                    self.cds = generic_xds_config["xdsConfig"]
+                elif re.search(r'\.ClusterLoadAssignment$',
+                               generic_xds_config['typeUrl']):
+                    self.eds = generic_xds_config["xdsConfig"]
+            except Exception as e:
+                logging.debug('Parse dumped xDS config failed with %s: %s',
```
Comment (gnossen) [inline tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_testcase.py:123]: Nit: `s/Parse/Parsing/`

File: src/python/grpcio_tests/tests/csds/test_csds.py (hunk 1 above)
Comment (lidizheng) [reply src/python/grpcio_tests/tests/csds/test_csds.py]: Removing this check, as this test case is meant for validating the Node information.

The CSDS implementation change happens in #27794.

There is no mutual exclusive guarantee, a server may duplicate same resource in both fields.

File: tools/run_tests/run_xds_tests.py (hunk 2 above)
Comment (lidizheng) [reply tools/run_tests/run_xds_tests.py]: Changed to `if not len(route_config['virtual_hosts']):`.

I was trying to be thorough, but this line is unnecessary unless CPython broke.

File: tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_testcase.py (hunk 3 above)
Comment (lidizheng) [reply tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_testcase.py]: Thanks! Done.

**Review by lidizheng — COMMENTED** [review]
Thanks for the quick review! I have addressed all comments.

**Pull request merged**